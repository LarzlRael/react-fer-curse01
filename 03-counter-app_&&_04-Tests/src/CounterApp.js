import React, { useState } from 'react'
import PropTypes from 'prop-types';

const CounterApp = ({ value = 10 }) => {


    //using the useState
    // first argument is the name of state 
    // second argument is the function to update the state

    const [counter, setCounter] = useState(value);


    const handleAdd = () => {
        //handleAdd

        // setCounter(counter+1)
        setCounter((c) => {
            return c + 1
        });

    }

    const handleSubstract = () => {
        //handleAdd

        setCounter(counter - 1)


    }
    const handleReset = () => {
        //handleAdd
        setCounter(0);
    }

    return (
        <div >
            <h1>CounterAPP</h1>
            <h2>{value}</h2>
            <h3>El nombre por defecto es : {counter}</h3>
            <button onClick={handleAdd}>+1</button>
            <button onClick={handleReset}>Reset</button>
            <button onClick={handleSubstract}>-1</button>


        </div>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number
}

CounterApp.defaultProps = {
    value: 10
}
export default CounterApp;
