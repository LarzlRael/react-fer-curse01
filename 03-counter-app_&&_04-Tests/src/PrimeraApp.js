import PropTypes from 'prop-types';
// Functional components
const PrimeraApp = ({ saludo, subtitulo }) => {



    const array = [1, 2, 3];

    const persona = {
        nombre: 'Larz',
        apellido: 'xd'
    }
    return (
        <>
            <h1>{saludo}</h1>
            {/* <pre>{JSON.stringify(persona,null,3)}</pre> */}
            <p>{subtitulo}</p>
            <span>hola mundo 2</span>
        </>
    )
}

// default properties
PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitulo: 'I am a default subtitle '
}
export default PrimeraApp;