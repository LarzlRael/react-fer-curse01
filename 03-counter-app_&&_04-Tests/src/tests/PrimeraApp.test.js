// import { render } from "@testing-library/react";
// import React from 'react';

import { shallow } from 'enzyme';
import CounterApp from '../CounterApp';
import PrimeraApp from "../PrimeraApp";

// describe('Testing in <PrimeraApp />', () => {

//     test('Debe mostrar el mensaje de hola soy Goku', () => {

//         const saludo = 'Hola Soy Goku';

//         const { getByText } = render(<PrimeraApp saludo={saludo} />);


//         expect(getByText(saludo)).toBeInTheDocument();

//     });

// });

describe('Testing in <PrimeraApp />', () => {

    // test('Debe mostrar el mensaje de hola soy Goku', () => {

    //     const saludo = 'Hola Soy Goku';

    //     const wraper = shallow(<PrimeraApp saludo={saludo} />);

    //     expect(wraper).toMatchSnapshot();
    // });

    let wrapper = shallow(<CounterApp />);

    // initialize components
    beforeEach(() => {
        wrapper = shallow(<CounterApp />); 
    })


    test('should show the subtitle sended by properties', () => {

        const saludo = 'Hola Soy Goku';
        const subTitulo = 'Soy un subtitulo';

        const wrapper = shallow(
            <PrimeraApp
                saludo={saludo}
                subtitulo={subTitulo}
            />
        );

        //Find paragraph element in wrapper or the PrimerApp
        const textParagraph = wrapper.find('p').text();
        //in textParagraph find the text inside the paragraph
        expect(textParagraph).toBe(subTitulo);

    });


    test('should be testing CounterApp', () => {

        const value = 1;

        const wraper = shallow(<CounterApp value={value} />);

        expect(wraper).toMatchSnapshot();
    });


    test('must show by default value in CounterApp', () => {

        const defaultValue = 100;
        const wrapper = shallow(
            <CounterApp value={100} />
        );

        //Find paragraph element in wrapper or the PrimerApp
        const headTwo = wrapper.find('h2').text().trim();
        //in textParagraph find the text inside the paragraph
        expect(headTwo).toBe(defaultValue);

    });

    test('must increment counter +1', () => {

        // first button HandleAdd
        wrapper.find('button').at(0).simulate('click');

        const counterText = wrapper.find('h2').text().trim();
        console.log(counterText);
        expect(counterText).toBe('11');

    });

    test('must decrease counter -1', () => {

        // third button HandleAdd
        wrapper.find('button').at(2).simulate('click');

        const counterText = wrapper.find('h2').text().trim();
        console.log(counterText);
        expect(counterText).toBe('9');

    });
    test('Should be set defualt value', () => {

        const wrapper = shallow(
            <CounterApp value={105} />
        );
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');


        wrapper.find('button').at(1).simulate('click');

        const counterText = wrapper.find('h2').text().trim();
        console.log(counterText);
        expect(counterText).toBe('105');
    });



});

