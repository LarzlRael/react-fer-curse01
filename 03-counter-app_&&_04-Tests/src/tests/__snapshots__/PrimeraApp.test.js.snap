// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <PrimeraApp /> should be testing CounterApp 1`] = `
<div>
  <h1>
    CounterAPP
  </h1>
  <h2>
    1
  </h2>
  <h3>
    El nombre por defecto es : 
    1
  </h3>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</div>
`;
